### PR TITLE
Pass URL to ostreeuploader module for token authentication

### PR DESCRIPTION
**Pull request by doanac**
The token authentication works well but is tied to api.fio. This change allows is to use the `-server` option the user was already able to set from the CLI.

While changing these files my editor also re-organized the imports in a more standard format.

### CODE REVIEW

**Review by vkhoroz — COMMENTED** [review]
LGTM

File: pkg/ostreeuploader/push.go [hunk 1]
```diff
@@ -120,19 +116,20 @@ func (t FioToken) SetAuthHeader(req *http.Request) {
...
 }
 
-func newOSTreeHubAccessorWithToken(repo string, factory string, token string, apiVer string) (*ostreeHubAccessor, error) {
+func newOSTreeHubAccessorWithToken(repo, hubURL, factory, token, apiVer string) (*ostreeHubAccessor, error) {
 	if err := checkRepoDir(repo); err != nil {
 		return nil, err
 	}
 	hub := &OSTreeHub{
-		URL:     FioApiBaseUrl,
+		URL:     hubURL,
```
Comment (mike-sul) [inline pkg/ostreeuploader/push.go:124]: I think it shouldn't impact customer since we still use the creds archive in meta-lmp?

**Review by mike-sul — APPROVED** [review]

File: pkg/ostreeuploader/push.go (hunk 1 above)
Comment (doanac) [reply pkg/ostreeuploader/push.go:124]: that was my thinking. There will be work to use this correctly in ci-scripts.

Comment (mike-sul) [reply pkg/ostreeuploader/push.go:124]: There is also this doc https://docs.foundries.io/latest/user-guide/custom-ci/custom-ci-for-rootfs.html#push-ostree-repo-to-cloud that most likely is impacted by this change.

Maybe we should default to `"https://api.foundries.io"` if `hubURL` is not specified.

Comment (doanac) [reply pkg/ostreeuploader/push.go:124]: i think i may have messed that up. i'll test this more today

Comment (doanac) [reply pkg/ostreeuploader/push.go:127]: I did just do a fixup change to make our URL concatenation/joining logic safer.

**Pull request merged**